File: C++/First Steps in Coding - Basics of C++/Nested Loops/Exercise/Sum Prime Non Prime.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
	int primeSum = 0, nonPrimeSum = 0;
	while (true) {
		string number;
		getline(cin, number);
		if (number != "stop") {
			int num = stoi(number);
			bool prime = false;
			if (num < 0) {
				cout << "Number is negative." << endl;
				continue;
			}
			for (int k = 1; k <= 10; k++) {
				if (k == 1 or k == num) {
					continue;
				}
				if (num%k == 0) {
					prime = true;
					break;
				}
			}
			if (prime == false) {
				nonPrimeSum += num;
			}
			else {
				primeSum += num;
			}
		}
		else {
			cout << "Sum of all prime numbers is: " << nonPrimeSum << "\nSum of all non prime numbers is: " << primeSum;
			break;
		}
	}
}

File: Python/Basics/Nested Loops/Exercise/Sum Prime Non Prime.py
sumPrime = 0
sumNonPrime = 0

while True:
    n = input()

    if n != "stop":
        prime = True
        n = int(n)

        if n < 0:
            print("Number is negative.")
            continue

        for k in range(2, n):
            if n % k == 0:
                prime = False
                break

        if not prime:
            sumNonPrime += n
        else:
            sumPrime += n

    else:
        break

print(f"Sum of all prime numbers is: {sumPrime}")
print(f"Sum of all non prime numbers is: {sumNonPrime}")